### ISSUE TITLE
Endpoint that updates a user in mongoDB

### ISSUE THREAD

**Issue by NickPapacostas** [SocialHackersClass10/ChalleduApp#29]
Acceptance criteria:

Create a new PUT /users/:id endpoint
Update the document in mongodb
On success return 200 and 
```
{
    _id: <id>
}
```
If the :id is missing then return 404 and 
```
{
    error: 'resource not found' (or something similar)
}
```
On other errors return 500 and the above body

**Comment by student0b4dc0d3** [comment]
About the "approved" and "approval_pending" fields

My thoughts where that "approval_pending" would serve as the flag of: 
is this specific user "prevented" to login and interact with our app?

The value of this field would serve for both the methods you described on the issue, as follows
When 1st the user is created (during user registration) this fields has a **forced** value of **true**, up and until any user update from an admin is posted with this field marked specifically as "false", meaning that an admin has approved the registration and is green-lighting (validating) this account. The value only changes when specifically included in the AJAX request body, which is **only after an admin update** (at document creation this field should be forced true)

To ensure this, the front-end user modification form should offer this field for update only when an admin is using that form. When a non-admin-role is interacting with user-data-update, this field must be hidden and unavailable for changes

If you feel this would not suffice for our needs (or over-complicate handling) then I have no objection implementing a second field. I just feel that we can handle requirements with a singular point of approval.

I would love to hear yours and the Team's thoughts on the matter.
Cheers and have an awesome day

**Comment by zannis** [comment]
Ok so the way I though about this was that the admin would be able to approve or reject any profile that is on their "Pending approvals" page on the frontend. The new field would essentially remove the rejected profiles from that list. But at this point, its not that necessary. 
With my idea, when creating a new user, we wouldn't set this field at all, and only set this after a approval or rejection has been made. I agree that it might be redundant and having all pending profiles present on that list is fine for the time being. I am updating the Acceptance Criteria to reflect this.

**Comment by student0b4dc0d3** [comment]
Hmm... 

> With my idea, when creating a new user, we wouldn't set this field at all, and only set this after a approval or rejection has been made. I agree that it might be redundant and having all pending profiles present on that list is fine for the time being

I see and I understand, and I also agree, if we don't implement the 2nd filed the already rejected registrations will never get resolved.
So, in hindsight I totally agree, we need another field signifying the "rejected" state.
If you don;t mind, I would like to implement that

Cheers and have an awesome evening

**Comment by student0b4dc0d3** [comment]
Or, ( even better imho ) we can instead transform the already existing field into non-binary enum named registration_status = enum [ "accepted" , "pending" , "rejected" ]
Both in the users and the NGO's
How does that sound?

**Comment by student0b4dc0d3** [comment]
@zannis 
Please let me know if you require me to quickly modify the current Schema.js, to include above changes so that other teams can asap incorporate those changes into their tickets
ty

**Comment by student0b4dc0d3** [comment]
created new ticket for the above requirement
** Document approval - Schema modification #38 **
**https://github.com/SocialHackersClass10/ChalleduApp/issues/38**

created new branch
**issue#38-document-approval-schema-modification**

completed the requirements, committed the changes and awaiting review / comments / resolution
**https://github.com/SocialHackersClass10/ChalleduApp/pull/39**

ty
